cmd/dashboard.go

```diff
@@ -586,6 +582,324 @@ func (d *dashboardClusterPane) Render() error {
 	return nil
 }
 
+type dashboardClusterVersionsPane struct {
+	kubernetesVersions         *widgets.Tree
+	reconciledGardenerVersions *widgets.PieChart
+
+	kubernetesVersionsState *kubernetesVersions
+
+	sem *semaphore.Weighted
+
+	cache *cloudCache
+}
+
+func NewDashboardClusterVersionsPane(cache *cloudCache, eventChannel chan ui.Event) *dashboardClusterVersionsPane {
+	d := &dashboardClusterVersionsPane{}
+
+	d.sem = semaphore.NewWeighted(1)
+
+	d.reconciledGardenerVersions = widgets.NewPieChart()
+	d.reconciledGardenerVersions.Title = "Reconciled Gardener Versions"
+
+	d.kubernetesVersions = widgets.NewTree()
+	d.kubernetesVersions.Title = "Kubernetes Versions"
+	d.kubernetesVersions.TextStyle = ui.NewStyle(ui.ColorYellow)
+	d.kubernetesVersionsState = &kubernetesVersions{}
+
+	go func() {
+		for {
+			e := <-eventChannel
+
+			switch e.ID {
+			case "<Down>":
+				d.kubernetesVersions.ScrollDown()
+			case "<Up>":
+				d.kubernetesVersions.ScrollUp()
+			case "<Space>", "<Enter>":
+				d.kubernetesVersions.ToggleExpand()
+			case "<PageUp>":
+				d.kubernetesVersions.ScrollTop()
+			case "<PageDown>":
+				d.kubernetesVersions.ScrollBottom()
+			case "E":
+				d.kubernetesVersions.ExpandAll()
+			case "C":
+				d.kubernetesVersions.CollapseAll()
+			}
+
+			ui.Render(d.kubernetesVersions)
+		}
+	}()
+
+	d.cache = cache
+
+	return d
+}
+
+func (d *dashboardClusterVersionsPane) Name() string {
+	return "ClusterVersions"
+}
+
+func (d *dashboardClusterVersionsPane) Description() string {
+	return "Cluster version information"
+}
+
+func (d *dashboardClusterVersionsPane) Resize(x1, y1, x2, y2 int) {
+	d.kubernetesVersions.SetRect(x1, y1, x2-40, y2)
+	d.reconciledGardenerVersions.SetRect(x2-40, y1, x2, y2)
+}
+
+type gardenerVersions map[string]int
+
+func (g gardenerVersions) total() int {
+	var result int
+	for _, amount := range g {
+		result += amount
+	}
+	return result
+}
+
+func (g gardenerVersions) toValues() ([]float64, widgets.PieChartLabel) {
+	type gardenerVersion struct {
+		parsedVersion *semver.Version
+		rawVersion    string
+		percentage    float64
+	}
+
+	var versions []gardenerVersion
+
+	for rawVersion, count := range g {
+		v, _ := semver.NewVersion(rawVersion)
+
+		versions = append(versions, gardenerVersion{
+			parsedVersion: v,
+			rawVersion:    rawVersion,
+			percentage:    float64(count) / float64(g.total()),
+		})
+	}
+
+	sort.Slice(versions, func(i, j int) bool {
+		if versions[i].parsedVersion == nil && versions[j].parsedVersion == nil {
+			return versions[i].rawVersion < versions[j].rawVersion
+		}
+
+		if versions[i].parsedVersion == nil {
+			return false
+		}
+		if versions[i].parsedVersion == nil {
```

Comment:
```suggestion
		if versions[j].parsedVersion == nil {
```

Reply:
Why not:

```
sort.Sort(semver.Collection(slice-of-semvers))
```
?

Reply:
It should also sort non-parsable versions by string.

Reply:
Don't know if there is a chance to have something like SHAs in there as we have it for the firewall-controller for instance. Just wanted to make sure we always have a defined sort order. 😅